Safari shows black UI after entering Portal Mode for MSE spatial Videos.

jyavenard opened the pull request:
#### 93215ecea1e953f8bdcb85bb1faf7e08513f9600
<pre>
Safari shows black UI after entering Portal Mode for MSE spatial Videos.
<a href="https://bugs.webkit.org/show_bug.cgi?id=294974">https://bugs.webkit.org/show_bug.cgi?id=294974</a>
<a href="https://rdar.apple.com/153932317">rdar://153932317</a>

Reviewed by Jer Noble.

This is a regression from 295444@main. Following this change, we copied
the spatial metadata to the decoded CMSampleBuffer, however it would now
indicate that it has both a left and right image but we only provided the
left eye image.

For now, on visionOS, if the media being played contains spatial or immersive
content, disable WebCoreDecompressionSession and fallback to using AVSBDL.

* Source/WebCore/platform/graphics/cocoa/VideoMediaSampleRenderer.h: When enqueuing a compressed sample,
also store if decompression session is to be used.
* Source/WebCore/platform/graphics/cocoa/VideoMediaSampleRenderer.mm:
(WebCore::VideoMediaSampleRenderer::enqueueSample): Check if content has spatial or immersive content,
set flag accordingly.
(WebCore::VideoMediaSampleRenderer::decodeNextSampleIfNeeded):
(WebCore::VideoMediaSampleRenderer::totalVideoFrames const): use renamed member.
(WebCore::VideoMediaSampleRenderer::droppedVideoFrames const): use renamed member.
(WebCore::VideoMediaSampleRenderer::corruptedVideoFrames const): use renamed member.
(WebCore::VideoMediaSampleRenderer::totalFrameDelay const): use renamed member.

Canonical link: <a href="https://commits.webkit.org/296748@main">https://commits.webkit.org/296748@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8d2e19c63bbb51a584b1a6ac4dee21da54e36112

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/109426 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/29084 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/19513 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/114631 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/59659 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/111389 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/29763 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/37672 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/83170 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/112374 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/23705 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/98559 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/63628 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/23086 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/16702 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/59251 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/93073 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/16744 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/117745 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/36467 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/26996 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/92183 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/36838 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/94821 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/91998 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23433 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/36926 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/14670 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/32271 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/36360 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/41836 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/36031 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/39370 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/37735 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/8d2e19c63bbb51a584b1a6ac4dee21da54e36112. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/47295

jyavenard committed:
Safari shows black UI after entering Portal Mode for MSE spatial Videos.
https://bugs.webkit.org/show_bug.cgi?id=294974
rdar://153932317

Reviewed by Jer Noble.

This is a regression from 295444@main. Following this change, we copied
the spatial metadata to the decoded CMSampleBuffer, however it would now
indicate that it has both a left and right image but we only provided the
left eye image.

For now, on visionOS, if the media being played contains spatial or immersive
content, disable WebCoreDecompressionSession and fallback to using AVSBDL.

* Source/WebCore/platform/graphics/cocoa/VideoMediaSampleRenderer.h: When enqueuing a compressed sample,
also store if decompression session is to be used.
* Source/WebCore/platform/graphics/cocoa/VideoMediaSampleRenderer.mm:
(WebCore::VideoMediaSampleRenderer::enqueueSample): Check if content has spatial or immersive content,
set flag accordingly.
(WebCore::VideoMediaSampleRenderer::decodeNextSampleIfNeeded):
(WebCore::VideoMediaSampleRenderer::totalVideoFrames const): use renamed member.
(WebCore::VideoMediaSampleRenderer::droppedVideoFrames const): use renamed member.
(WebCore::VideoMediaSampleRenderer::corruptedVideoFrames const): use renamed member.
(WebCore::VideoMediaSampleRenderer::totalFrameDelay const): use renamed member.

Canonical link: https://commits.webkit.org/296748@main
